lib/UserAgentApplication.ts

```diff
@@ -585,22 +630,77 @@ namespace Msal {
+        acquireTokenPopup(scopes: Array<string>, authority: string): Promise<string>;
+        acquireTokenPopup(scopes: Array<string>, authority: string, user: User): Promise<string>;
+        acquireTokenPopup(scopes: Array<string>, authority: string, user: User, extraQueryParameters: string): Promise<string>;
+        acquireTokenPopup(scopes: Array<string>, authority?: string, user?: User, extraQueryParameters?: string): Promise<string> {
+            return new Promise<string>((resolve, reject) => {
+                const isValidScope = this.validateInputScope(scopes);
+                if (isValidScope && !Utils.isEmpty(isValidScope)) {
+                    reject(isValidScope);
+                }
+
+                const userObject = user ? user : this.user;
+                if (this._acquireTokenInProgress) {
+                    reject("AcquireToken is in progress");
+                    return;
+                }
+
+                const scope = scopes.join(" ").toLowerCase();
+                if (!userObject) {
+                    reject("user login is required");
+                    return;
+                }
+
+                this._acquireTokenInProgress = true;
+                let authenticationRequest: AuthenticationRequestParameters;
+                const acquireTokenAuthority = authority ? authority : this.authority;
+                if (Utils.compareObjects(userObject, this.user)) {
+                    authenticationRequest = new AuthenticationRequestParameters(acquireTokenAuthority, this.clientId, scopes, ResponseTypes.token, this.redirectUri);
+                } else {
+                    authenticationRequest = new AuthenticationRequestParameters(acquireTokenAuthority, this.clientId, scopes, ResponseTypes.id_token_token, this.redirectUri);
+                }
+
+                this._cacheStorage.setItem(Constants.nonceIdToken, authenticationRequest.nonce);
+                authenticationRequest.state = authenticationRequest.state + "|" + scope;
+                const acquireTokenUserKey = Constants.acquireTokenUser + Constants.resourceDelimeter + userObject.userIdentifier + Constants.resourceDelimeter + authenticationRequest.state;
+                if (Utils.isEmpty(this._cacheStorage.getItem(acquireTokenUserKey))) {
+                    this._cacheStorage.setItem(acquireTokenUserKey, JSON.stringify(userObject));
+                }
+
+                const authorityKey = Constants.authority + Constants.resourceDelimeter + authenticationRequest.state;
+                if (Utils.isEmpty(this._cacheStorage.getItem(authorityKey))) {
+                    this._cacheStorage.setItem(authorityKey, acquireTokenAuthority);
+                }
+
+                if (extraQueryParameters) {
+                    authenticationRequest.extraQueryParameters = extraQueryParameters;
+                }
+
+                let urlNavigate = authenticationRequest.createNavigateUrl(scopes) + "&prompt=select_account";
```

Comment:
>&prompt=select_account [](start = 85, length = 22)

Non blocking: do you agree that ideally this function should add the prompt=select_account qp?